internal/distro/rhel86/pipelines.go

```diff
@@ -52,32 +52,47 @@ func prependKernelCmdlineStage(pipeline *osbuild.Pipeline, t *imageType, pt *dis
 	return pipeline
 }
 
-func vhdPipelines(t *imageType, customizations *blueprint.Customizations, options distro.ImageOptions, repos []rpmmd.RepoConfig, packageSetSpecs map[string][]rpmmd.PackageSpec, rng *rand.Rand) ([]osbuild.Pipeline, error) {
-	pipelines := make([]osbuild.Pipeline, 0)
-	pipelines = append(pipelines, *buildPipeline(repos, packageSetSpecs[buildPkgsKey], t.arch.distro.runner))
+func vhdPipelines(compress bool) pipelinesFunc {
```

Comment:
This is cool.  We should do more of this.

Reply:
I agree that this is clever, but not sure if we do more of this 🙄  The only reason from my PoV is that I'm not sure how it will play out once we at some point start pulling definitions to more declarative form. I feel like these kind of things will make it more complicated in the future.

We do a very similar thing, but in a different way for the `ec2` and `ami` images.

Reply:
More descriptive pipelines would be very nice but I think it will need quite a bit of (re-)design work on various levels.